tests/pytests/functional/cli/test_api.py

```diff
@@ -0,0 +1,34 @@
+import os
+import sys
+import tempfile
+
+import pytest
+from salt.cli.api import SaltAPI
+from tests.support.mock import patch
+
+
+@pytest.mark.slow_test
+def test_start_shutdown():
+    pidfile = tempfile.mktemp()
+    logfile = tempfile.mktemp()
+    api = SaltAPI()
+    with pytest.raises(SystemExit):
+        try:
+            # testing environment will fail if we use default pidfile
+            # overwrite sys.argv so salt-api does not use testing args
+            with patch.object(
+                sys, "argv", [sys.argv[0], "--pid-file", pidfile, "--log-file", logfile]
+            ):
+                api.start()
+                assert os.path.isfile(pidfile)
+                assert os.path.isfile(logfile)
```

Comment:
should this check for api_logfile?

Reply:
No `logfile` is correct, its the name of the temporary file we're passing to `--log-file`.

Reply:
I'm confused. 

`salt/cli/api.py` now uses `api_logfile` instead of `log_file`, that's fine.

But where is `api_logfile` assigned to the value of `log_file`? 

The closest thing I'm seeing is that the reverse happens - `salt/config/__init__.py:api_config` sets `log_file` to the value of `api_logfile`.

Shouldn't this test at least verify that `api_logfile` exists, as it's what `salt/cli/api.py` should be using?

Looking at the code it really seems like the default `api_logfile` is still being used, maybe the reason this test passes is because the test framework has access to `$(root_dir)/api`?

Reply:
Ok, thanks for the clarity - to sum up:

SaltAPIParser has https://github.com/saltstack/salt/blob/3f3010415d7e12b9bfad15efa592ed024e314a6a/salt/utils/parsers.py#L3584

Which is used in a mixin to set the location `--log-level` is stored.

https://github.com/saltstack/salt/blob/3f3010415d7e12b9bfad15efa592ed024e314a6a/salt/utils/parsers.py#L592-L595